src/main/java/core/basesyntax/impl/StorageImpl.java

```
@@ -3,17 +3,64 @@
 import core.basesyntax.Storage;
 
 public class StorageImpl<K, V> implements Storage<K, V> {
+    public static final int SIZE_OF_STORAGE = 10;
+    private final Cargo<K, V>[] cargos;
+
+    public StorageImpl() {
+        this.cargos = (Cargo<K, V>[]) new Cargo[SIZE_OF_STORAGE];
+    }
+
     @Override
     public void put(K key, V value) {
+        int index = getIndexOfKey(key);
+
+        if (index >= 0) {
+            cargos[index].setValue(value);
+        } else {
+            for (int i = 0; i < SIZE_OF_STORAGE; i++) {
+                if (cargos[i] == null) {
+                    cargos[i] = new Cargo(key, value);
+                    return;
+                }
+                System.out.println("The storage is full!");
```

This violates checklist item: "When capacity is exceeded (attempt to put beyond 10), the code must have a defined, consistent behavior (e.g., ignore, overwrite, or throw a runtime exception). This must be implemented explicitly since capacity is constrained to 10 by arrays; behavior must be verifiable in code."

Currently, the code prints a message for every non-null slot, which is not correct. Instead, after the for-loop, if no empty slot is found, you should throw an exception (e.g., IllegalStateException) to make the overflow behavior explicit and verifiable.